### DISCUSSION TITLE
Is there a way to close all open popup windows?

### DISCUSSION THREAD

**Discussion by jyapayne** [mfussenegger/nvim-dap#294] [Q&A]
This would be especially useful so that I could map it to a key combo in vim instead of having to type `:q` all the time.

**Comment by mfussenegger** [comment]
`CTRL-w q` is the default keybinding in nvim to close a window and it also works on popups and you could remap it if you want.

Or are you looking for something else?

**Comment by jyapayne** [comment]
I was looking for something quick that also wouldn't close the current buffer if I accidentally hit it twice.

**Reply by mfussenegger** [reply]
There is also `:only` / `CTRL-w o`

**Reply by jyapayne** [reply]
Looks like that doesn't work :( I get the error "Cannot close window, only floating window would remain". I can do something like this, but it's really slow:

```vim
:lua for _, win in ipairs(vim.api.nvim_list_wins()) do local config = vim.api.nvim_win_get_config(win); if config.relative ~= "" then vim.api.nvim_win_close(win, false); print('Closing window', win) end end
```

It would be great to have a lua function in nvim-ldap that just closes all floating windows.

**Reply by jyapayne** [reply]
Actually it's not slow, just had a delay due to bindings waiting for input. But it still would be nice to just have a nvim-dap function to do it

**Reply by mfussenegger** [reply]
Seems a bit out-of-scope as it is general window management. There is nothing nvim-dap specific in your command.

**Reply by jyapayne** [reply]
I was thinking of something that just closes the open floating windows that nvim-dap has opened. Not any other windows. Would it be out of scope to track the ones that nvim-dap have opened and just have a function to close them?

**Reply by mfussenegger** [reply]
Maybe, not sure. It may be a bit tricky because there are some functions that allow users to completely control how/if windows are created (for the integrated terminal, but also via the widget API)

I may accept a PR if it isn't introducing too much complexity.
